pkg/storage/tsdb/bucketcache/caching_bucket.go

```diff
@@ -159,6 +159,38 @@ func NewCachingBucket(bucketID string, bucketClient objstore.Bucket, cfg *Cachin
 	return cb, nil
 }
 
+func (cb *CachingBucket) invalidate(ctx context.Context, name string) {
```

Comment:
Should we allow to conditionally enable invalidation, so that we enable it only for caching buckets where we need it? Not feeling strong about it.

Reply:
I added an option to `CachingBucketConfig::ConfigGet()`